src/test/java/net/snowflake/client/jdbc/FileUploaderExpandFileNamesTest.java

```diff
@@ -126,4 +138,74 @@ public int read() throws IOException {
     assertEquals("dummy_dest_file_name", config.getDestFileName());
     assertEquals(expectedThrowCount, throwCount);
   }
+
+  /**
+   * We have N jobs expanding files with exclusive pattern, processing them and deleting. Expanding
+   * the list should not cause the error when file of another pattern is deleted which may happen
+   * when FileUtils.listFiles is used.
+   *
+   * <p>Fix available after version 3.16.1.
```

Comment:
I will be next version after 3.16.1?

Reply:
we haven't released yet so this statement is still true (and we are planning to include this fix in the next release)